topeft/modules/datacard_tools.py

```diff
@@ -422,7 +422,8 @@ def read(self,fpath):
         for km_dist, h in self.hists.items():
             if h.empty():
                 continue
-            if self.var_lst and km_dist not in self.var_lst:
+            km_dist_sumw = km_dist.replace('_sumw2','')
+            if self.var_lst and km_dist not in self.var_lst and km_dist_sumw not in self.var_lst:
```

Comment:
So now with the restructured `if` check, I'm wondering what this last part of the boolean check is actually looking for? I read it as: 1st we want to make sure that `self.var_lst` is non-empty, 2nd we want to make sure that `km_dist` is not an element of that list, but then I'm not sure I understand what the 3rd check is looking for. It sounds to me like `self.hists,items()` will contain both the strings `"some_dist"` and `"some_dist_sumw2"`, which I think is fine, but it gives me pause when reading this line, since it means we're splitting how the `for` loop works depending on if a certain distribution exists _and_ if it also has this new `sumw2` histogram.

What you have here might make total sense and we can leave it as is, but maybe you can try to explain the thought process here to help me understand what we're checking for.

Reply:
What I'm doing here is only letting it load the original `km_dist` _and_ the `km_dist_suw2`. Everything else is skipped.

Reply:
I wonder if it would make it easier to read and more self documenting to have something like:
```
km_dist_missing = km_dist not in self.var_lst
km_base_dist_missing = km_dist.replace('_sumw2','')
if self.var_lst and km_dist_missing and km_base_dist_missing:
```
But I'd still like to hear you're interpretation of what this check is supposed to be doing

Reply:
I'm not sure I understand what you mean about letting it load _only_ the original and the `sumw2` version. Wouldn't any "original" distribution that shows up in `self.var_lst` pass this check regardless if it has a `sumw2` complement?

Reply:
I'd maybe describe this check instead as "We only want to include a `sumw2` histogram if it's corresponding "base" `km_dist` is part of the `self.var_lst` list". Is that a reasonable interpretation?

Reply:
Yes, the original distribution always passes, but the `sumw2` histogram doesn't pass by default because e.g. `lj0pt_sumw2` _isn't_ in `self.var_lst`

Reply:
> I wonder if it would make it easier to read and more self documenting to have something like:
> 
> ```
> km_dist_missing = km_dist not in self.var_lst
> km_base_dist_missing = km_dist.replace('_sumw2','') not in self.var_lst
> if self.var_lst and km_dist_missing and km_base_dist_missing:
> ```
> 
> But I'd still like to hear you're interpretation of what this check is supposed to be doing (also I realize I'm being a hypocrite regarding my earlier comment about not having `bool` checks do multiple things at the same time here)

I like this suggestion, it is a bit easier to read.